adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -294,22 +294,24 @@ public virtual void Clear()
         {
             lock (cacheLock)
             {
+                TokenCacheNotificationArgs args = new TokenCacheNotificationArgs { TokenCache = this };
+                this.OnBeforeAccess(args);
```

Comment:
is it needed to have the lock on the cache for the onbeforeaccess?

Reply:
probably no , in general onBeforeAccess call is not necessary here , we do not need to read it here.
But It is general pattern of accessing cache here (in this lib)

Reply:
would have liked to see if we could loosen this a bit and have fewer locks

---
In reply to: [223477053](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r223477053) [](ancestors = 223477053)